src/app/team.ts

```diff
@@ -15,23 +15,59 @@
 export class Team {
     constructor(
         public id: string,
-        public displayName: string
+        public displayName: string,
+        public teamPermissions?: TeamPermissions
     ) {}
 }
 
+export class TeamList {
+  constructor(
+    public teams: Team[],
+    public addTeamPermissions: boolean
+  ) {}
+}
+
+export class TeamPermissions {
+  constructor(
+    public read: Permission,
+    public write: Permission,
+    public readAllTeams: Permission,
+    public addTeam: Permission,
+  ) {
+  }
+}
+
+export class Permission {
+  constructor(
+    public allow: Principal[]
+  ) {
+  }
+}
+
+export class Principal {
```

Comment:
Can we rename this to be consistent with what we did in the backend?

Reply:
👍🏻